clue.js

```diff
@@ -0,0 +1,316 @@
+
+//GROUPING ALL SUSPECTS, WEAPONS AND ROOMS IN ARRAYS
+
+const suspects = [
+  mrGreen,
+  mrsWhite,
+  professorPlum,
+  missScarlet,
+  mrsPeacock,
+  colonelMustard
+]
+
+const weapons = [
+  rope,
+  knife,
+  candlestick,
+  dumbbell,
+  poison,
+  axe,
+  bat,
+  trophy,
+  pistol
+]
+
+const rooms = [
+  diningRoom,
+  conservatory,
+  kitchen,
+  study,
+  library,
+  billiardRoom,
+  lounge,
+  ballroom,
+  hall,
+  spa,
+  livingRoom,
+  observatory,
+  theater,
+  guestHouse,
+  patio
+]
+
+// THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT YOU PASS IN TO THE FUNCTION.
+
+const randomSelector = array => {
+  return array[Math.floor(Math.random() * array.length)]
+}
+
+// CREATE AN OBJECT THAT KEEPS THE MYSTERY.
+
+let mysteryobject = {
+  killer: "",
+  weapon: "",
+  room: "",
+}
+
+// This function will be invoked when you click on the killer card.
+const pickKiller = () => {
+  // This will randomly select a killer from the suspects. And add that to the mystery object.
+  mystery.killer = randomSelector(suspects)
+
+  // This will change the background color of the card to the one connected to the chosen killer and show selected elements
+  document.getElementById('killerCard').style.background = mystery.killer.color
+  document.getElementById('killerName').innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+  document.getElementById('killerImage').src = `${mystery.killer.image}`
+  document.getElementById('killerAge').innerHTML = `Age: ${mystery.killer.age}`
+  document.getElementById('killerOccupation').innerHTML = `Occupation: ${mystery.killer.occupation}`
+  document.getElementById('killerDescription').innerHTML = `Description: ${mystery.killer.description}`
+  document.getElementById("killerFavouriteWeapon").innerHTML = `Favourite weapon: ${shuffleFavouriteWeapon()}`
+}
+
+document.getElementById('killerCard').addEventListener('click', pickKiller)
+
+// CREATE FUNCTIONS pickWeapon 
+
+const pickWeapon = () => {
+  mystery.weapon = randomSelector(weapons)
+
+  document.getElementById("weaponName").innerHTML = `${mystery.weapon.name}`
+  document.getElementById("weaponWeight").innerHTML = `Weight: ${mystery.weapon.weight} kilos`
+  document.getElementById("weaponImage").src = mystery.weapon.image
+}
+
+document.getElementById('weaponCard').addEventListener('click', pickWeapon)
+
+
+// CREATE FUNCTIONS pickRoom 
+
+const pickRoom = () => {
+  mystery.room = randomSelector(rooms)
+
+  document.getElementById("roomName").innerHTML = `${mystery.room.name}`
+  document.getElementById("roomImage").src = mystery.room.image
+}
+
+document.getElementById('roomCard').addEventListener('click', pickRoom)
+
+
+// Shuffle killers favourite weapon
+const shuffleFavouriteWeapon = () => {
+  return randomSelector(weapons).id
+}
+
+
+// CREATE A FUNCTION revealMystery that will be invoked when you click that button. 
+
+const revealMystery = () => {
+  if (!mystery.killer || !mystery.weapon || !mystery.room) {
```

Comment:
This is good, to not show any mystery object info before all the cards are picked :+1:

Reply:
And great if statement, good job Johanna!